public/diagnostic.js

```diff
@@ -23,3 +23,20 @@
 //  -  Order 1 : 2 hats ($5 each)
 //  -  Order 2 : 1 desk lamp ($20 each)
 //  -  Order 3 : 3 hand towels ($8 each)
+
+const Order = Ember.Object.extend ({
+  orderPrice: Ember.computed('unitPrice', 'quantity', function(){
+    return $this.get('unitPrice') * $this.get('quantity');
+  })
+});
+
+const Cart = Embed.object.extend ({
+  orders
```

Comment:
these two lines strike me more as pseudocode  _good_ pseudocode for the record. Completed I would expect something more like this:

```js
orders: [],
  addToCart: function(order){
    this.get('orders').addObject(order);
  },
```